### PR TITLE
ZBUG-982: Added code to check & display google calendar option only when its configured.

### PR DESCRIPTION
**Pull request by madhavdhuppe**

### CODE REVIEW

File: WebRoot/js/zimbraMail/calendar/controller/ZmCalendarTreeController.js [hunk 1]
```diff
@@ -538,10 +538,19 @@ function() {
...
 ZmCalendarTreeController.prototype._addExternalCalendarListener =
 function() {
-	var dialog = this.getExternalCalendarDialog();
-    dialog.popup();
+	return AjxRpc.invoke(null, "/service/extension/oauth2/info/google", null, new AjxCallback(this, this._handleGoogleConfig), true);
 };
 
+ZmCalendarTreeController.prototype._handleGoogleConfig =
+function(response){
+	var responseJson = JSON.parse(response.text);
```
Comment (silentsakky) [inline WebRoot/js/zimbraMail/calendar/controller/ZmCalendarTreeController.js:546]: JSON.parse inside try/catch block

File: WebRoot/js/zimbraMail/calendar/controller/ZmCalendarTreeController.js [hunk 2]
```diff
@@ -523,10 +523,10 @@ function(params) {
 * Returns an "External Calendar" dialog.
 */
 ZmCalendarTreeController.prototype.getExternalCalendarDialog =
-function() {
+function(isGooglAuthConfigured) {
```
Comment (silentsakky) [inline WebRoot/js/zimbraMail/calendar/controller/ZmCalendarTreeController.js:526]: typo: isGoogl**e**AuthConfigured

File: WebRoot/js/zimbraMail/calendar/controller/ZmCalendarTreeController.js (hunk 1 above)
Comment (madhavdhuppe) [reply WebRoot/js/zimbraMail/calendar/controller/ZmCalendarTreeController.js]: updated

File: WebRoot/js/zimbraMail/calendar/controller/ZmCalendarTreeController.js (hunk 2 above)
Comment (madhavdhuppe) [reply WebRoot/js/zimbraMail/calendar/controller/ZmCalendarTreeController.js]: updated

**Review by auminpatel — APPROVED** [review]

File: WebRoot/js/zimbraMail/calendar/controller/ZmCalendarTreeController.js [hunk 3]
```diff
@@ -538,10 +538,25 @@ function() {
...
+		var responseJson = JSON.parse(response.text);
+		let isGoogleAuthConfigured = false;
+
+		if(responseJson.data && responseJson.data.client_id){
+			isGoogleAuthConfigured = true;
+		}
+
+		var dialog = this.getExternalCalendarDialog(isGoogleAuthConfigured);
+		dialog.popup();
+	} catch(e) {
```
Comment (silentsakky) [inline WebRoot/js/zimbraMail/calendar/controller/ZmCalendarTreeController.js:556]: we don't need try/catch for all statements, just JSON.parse

Comment (madhavdhuppe) [reply WebRoot/js/zimbraMail/calendar/controller/ZmCalendarTreeController.js:556]: ok

File: WebRoot/js/zimbraMail/calendar/controller/ZmCalendarTreeController.js [hunk 4]
```diff
@@ -538,10 +538,25 @@ function() {
...
+function(response){
+	try {
+		var responseJson = JSON.parse(response.text);
+		let isGoogleAuthConfigured = false;
+
+		if(responseJson.data && responseJson.data.client_id){
+			isGoogleAuthConfigured = true;
+		}
+
+		var dialog = this.getExternalCalendarDialog(isGoogleAuthConfigured);
```
Comment (silentsakky) [inline WebRoot/js/zimbraMail/calendar/controller/ZmCalendarTreeController.js:554]:
```suggestion
		var dialog = this.getExternalCalendarDialog(responseJson.data && responseJson.data.client_id);
```

File: WebRoot/js/zimbraMail/calendar/controller/ZmCalendarTreeController.js [hunk 5]
```diff
@@ -538,10 +538,20 @@ function() {
...
 
+ZmCalendarTreeController.prototype._handleGoogleConfig =
+function(response){
+	try {
+		var responseJson = JSON.parse(response.text);
+	} catch(e) {
+		console.error("unable to handle Google Config get info response => ", e);
+	}
+
+	var dialog = this.getExternalCalendarDialog(responseJson.data && responseJson.data.client_id);
```
Comment (silentsakky) [inline WebRoot/js/zimbraMail/calendar/controller/ZmCalendarTreeController.js:552]: better to check for existence of responseJson also
when service returns invalid json at that time responseJson is undefined and accessing responseJson.data will raise javascript error

Comment (madhavdhuppe) [reply WebRoot/js/zimbraMail/calendar/controller/ZmCalendarTreeController.js:552]: ok will add it for more safety i already tested this on when configured || not configured

**Review by silentsakky — APPROVED** [review]

**Pull request merged**